apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -101,27 +98,20 @@ private static void AssertAttempts(RetryOptions retryOptions, int statusCode, in
         var client = new StorageClientImpl(service, encryptionKey: null, scheduler);
         scheduler.Run(() =>
         {
-            // Retries for the the failures. Assumed that error code 504 is always included in the predicate for these tests.
-            service.ExpectRequests(request, (HttpStatusCode) 504, numOfFailures - 1);
-            if (numOfFailures > 0)
+            for (int i = 0; i < requestStatusCodes.Length; ++i)
             {
-                service.ExpectRequest(request, (HttpStatusCode) statusCode);
+                service.ExpectRequest(request, (HttpStatusCode) requestStatusCodes[i]);
             }
 
             DateTime startTime = clock.GetCurrentDateTimeUtc();
             if (success)
             {
-                // Last call is a success.
-                service.ExpectRequest(request, new Bucket());
-
                 client.GetBucket("bucket", new GetBucketOptions { RetryOptions = retryOptions });
-
                 Assert.Equal(expectedBackOffs.Count(), messageHandler.AttemptTimestamps.Count());
                 service.Verify();
             }
             else
             {
-                // The call throws an exception
```

Comment:
We should still be calling service.Verify and asserting the expected backoffs even in the failure scenario.

`success` should *only* be used to determine whether the call eventually succeeds or not.

Reply:
> Tests will fail because in these cases all values of _requestResponses ( from ReplayingMessageHandler) are not dequeued as we are not consuming all loaded Status Codes.

So change the status codes provided by the test.

If you don't expect the client to retry at all, you should only provide a single status code.

We know exactly what the client should do, how many requests it should make, and the time between them. So that's what the mock needs to be "preprogrammed" with, and then we should verify that's exactly what happens.

Reply:
Made the required changes. Kindly verify.